packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -605,21 +606,27 @@ export class TreeCheckout implements ITreeCheckoutFork {
 					revertible.dispose();
 				}
 			},
-			clone: (forkedBranch: TreeBranch) => {
-				if (forkedBranch === undefined) {
-					return this.createRevertible(revision, kind, checkout, onRevertibleDisposed);
-				}
-
+			clone: (targetBranch: TreeBranch) => {
 				// TODO:#23442: When a revertible is cloned for a forked branch, optimize to create a fork of a revertible branch once per revision NOT once per revision per checkout.
-				const forkedCheckout = getCheckout(forkedBranch);
+				const targetCheckout = getCheckout(targetBranch);
+
 				const revertibleBranch = this.revertibleCommitBranches.get(revision);
-				assert(
-					revertibleBranch !== undefined,
-					0xa82 /* change to revert does not exist on the given forked branch */,
-				);
-				forkedCheckout.revertibleCommitBranches.set(revision, revertibleBranch.fork());
+				if (revertibleBranch === undefined) {
+					throw new UsageError("Change to revert does not exist on the branch.");
```

Comment:
This is not the error message I suggested in my comment.
```suggestion
					throw new UsageError("Unable to clone a revertible that has been disposed.");
```

Reply:
Ah, now I see why you suggested this error message. I was more obsessed with `Revertible` not existing in the branch but that in other words mean that it has been already disposed and thus not exist on the branch right?